sdk/core/core-util/package.json

```diff
@@ -62,12 +62,13 @@
   "prettier": "@azure/eslint-plugin-azure-sdk/prettier.json",
   "dependencies": {
     "@azure/abort-controller": "^2.0.0",
-    "tslib": "^2.2.0"
+    "tslib": "^2.2.0",
+    "type-plus": "^7.6.0"
```

Comment:
👋 I would strongly prefer not taking a runtime dependency here, and I think it's still in the guidelines https://azure.github.io/azure-sdk/typescript_implementation.html#ts-dependencies

Maybe we can relax the type system a bit here and still get type safety with some runtime validations? The only consumers of this API is Azure SDK developers, and we need runtime validations anyway (since of course none of the types exist at runtime). Can we rely on those and lean less heavily on the Type system?

Reply:
The functions are probably only going to be consumed internally in unit tests for merge-patch operations. The type definition, however, is meant to wrap the input type to DPG operations that follow the Merge Patch semantics. As a matter of principle, the DPG clients don't do any runtime validation of user input, so this is just meant to give users a hand using the type system.

We're having discussions about alternative approaches. If this ends up being the path forward for us, I'd be open to vendoring this dependency (MIT license). We've also considered putting more niche type definitions in their own package, reducing TS related bloat where it's not needed and allowing for customers to only depend on it at build time. 

Though, now that I think about it, it's only a compile-time dependency anyway. Not sure why I moved it out of the dev dependencies.